templates/postgresql-server-firewall-rules.json

```diff
@@ -20,23 +21,46 @@
       "metadata": {
         "description": "Name of the server to add firewall rules to"
       }
+    },
+    "subnetResourceIdList": {
+      "type": "array",
+      "defaultValue": [],
+      "metadata": {
+        "description": "A list of subnet resource ids."
+      }
     }
   },
   "variables": {},
   "resources": [
     {
-      "name": "[replace(concat(parameters('serverName'), '/', parameters('firewallRuleNamePrefix'), parameters('ipAddresses')[copyIndex()]), '.', '_')]",
+      "name": "[replace(concat(parameters('serverName'), '/', if(greater(length(parameters('firewallRuleNamePrefix')), 0) , parameters('ipAddresses')[copyIndex()], 'placeholder')), '.', '_')]",
+      "condition": "[greater(length(parameters('ipAddresses')), 0)]",
       "type": "Microsoft.DBforPostgreSQL/servers/firewallRules",
       "apiVersion": "2017-12-01",
       "properties": {
         "startIpAddress": "[parameters('ipAddresses')[copyIndex()]]",
         "endIpAddress": "[parameters('ipAddresses')[copyIndex()]]"
       },
       "copy": {
-        "count": "[length(parameters('ipAddresses'))]",
-        "mode": "Parallel",
-        "name": "firewallRuleCopy"
+        "count": "[if(greater(length(parameters('ipAddresses')), 0), length(parameters('ipAddresses')), 1)]",
```

Comment:
According to the documentation it's now possible to use `0` as a count in the `copy` properties, might be worth trying rather than that workaround https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-multiple#copy-limits